### PR TITLE
Fix up open ghost dir button a bit

### PR DESCRIPTION
**Pull request by stxticOVFL**
- Most likely fix the error where it displays "play next level" out of nowhere (there should be no reason why it happens anymore).
- Add it to the Job Archive and White's Journal:
image
image

### CODE REVIEW

**Comment by stxticOVFL** [comment]
Got a better idea that won't conceal the `Press [Space] to start` on different languages.

File: Modules/OpenGhostDir.cs [hunk 1]
```diff
@@ -10,50 +10,146 @@ internal class OpenGhostDir : Module
...
         private static MenuButtonHolder ghostButton;
+        private static bool jobArchive;
         private static MelonPreferences_Entry<bool> _setting_GhostButton;
 
         public OpenGhostDir() =>
             _setting_GhostButton = NeonLite.Config_NeonLite.CreateEntry("Open Ghost Directory Button", true, description: "Shows a button at the end to open this level's ghost directory in the file explorer.");
 
 
-        public static string GetGhostDirectory()
+        public static string GetGhostDirectory(LevelData level = null)
```
Comment (MOPSKATER) [inline Modules/OpenGhostDir.cs:20]: You can make this one private :3

File: Modules/OpenGhostDir.cs [hunk 2]
```diff
@@ -10,50 +10,146 @@ internal class OpenGhostDir : Module
...
+            if (level == null)
+                level = Singleton<Game>.Instance.GetCurrentLevel();
+            string path = GhostRecorder.GetCompressedSavePathForLevel(level);
             path = Path.GetDirectoryName(path) + "/";
             return path;
         }
 
+        private static void SetupButton(MenuButtonHolder button, LevelData level = null)
+        {
+            try
```
Comment (MOPSKATER) [inline Modules/OpenGhostDir.cs:31]: Try to put only what's needed in the try block. This improves readability

File: Modules/OpenGhostDir.cs [hunk 3]
```diff
@@ -10,50 +10,146 @@ internal class OpenGhostDir : Module
...
+            {
+                if (ghostButton != null)
+                {
+                    if (jobArchive)
+                    {
```
Comment (MOPSKATER) [inline Modules/OpenGhostDir.cs:36]: U are nesting very much here. Try ghostButton != null && jobArchive

File: Modules/OpenGhostDir.cs [hunk 4]
```diff
@@ -10,50 +10,146 @@ internal class OpenGhostDir : Module
...
+                        UnityEngine.Object.Destroy(ghostButton.gameObject);
+                        UnityEngine.Object.Destroy(ghostButton.transform.parent.gameObject);
+                        ghostButton = null;
+                    }
+                }
+
+                if (ghostButton == null)
```
Comment (MOPSKATER) [inline Modules/OpenGhostDir.cs:43]: There is propably a way to initialize ghostButton once.

File: Modules/OpenGhostDir.cs [hunk 5]
```diff
@@ -10,50 +10,146 @@ internal class OpenGhostDir : Module
...
+                UnityEngine.Debug.LogError("error on setting up the button " + e);
+            }
+        }
+
         [HarmonyPrefix]
         [HarmonyPatch(typeof(MenuScreenResults), "OnSetVisible")]
         private static void OnMenuScreen(ref MenuScreenResults __instance)
         {
-            if (ghostButton == null)
+            try
```
Comment (MOPSKATER) [inline Modules/OpenGhostDir.cs:77]: Try to put only what's needed in the try block. This improves readability

File: Modules/OpenGhostDir.cs [hunk 6]
```diff
@@ -10,50 +10,146 @@ internal class OpenGhostDir : Module
...
+                ghostButton.transform.parent.position = pos;
 
-                ghostButton.ButtonRef.onClick.RemoveAllListeners();
-                ghostButton.ButtonRef.onClick.AddListener(() => Process.Start("file://" + GetGhostDirectory()));
+                if (!__instance.buttonsToLoad.Contains(ghostButton))
+                    __instance.buttonsToLoad.Add(ghostButton);
             }
+            catch (Exception e)
+            {
+                UnityEngine.Debug.LogError("error on OnMenuScreen " + e);
```
Comment (MOPSKATER) [inline Modules/OpenGhostDir.cs:93]: For debug reasons I recommend e.printStackTrace(). It's way better

File: Modules/OpenGhostDir.cs [hunk 7]
```diff
@@ -10,50 +10,146 @@ internal class OpenGhostDir : Module
...
+                }
+
+                ghostButton.name = "Button Ghost";
+                ghostButton.buttonText = "Open Ghost Directory";
+                ghostButton.buttonTextRef.text = "Open Ghost Directory";
+                ghostButton.gameObject.SetActive(_setting_GhostButton.Value);
+            }
+            catch (Exception e)
+            {
+                UnityEngine.Debug.LogError("error on setting up the button " + e);
```
Comment (MOPSKATER) [inline Modules/OpenGhostDir.cs:69]: For debug reasons I recommend e.printStackTrace(). It's way better

**Review by MOPSKATER — CHANGES_REQUESTED** [review]
Your code is way to complex for just adding a single button. You should go back to analysis and try to really understand whats happening in the game. Then simplify your code as much as possible.

File: Modules/OpenGhostDir.cs (hunk 4 above)
Comment (stxticOVFL) [reply Modules/OpenGhostDir.cs:43]: As it currently stands, the `ghostButton` variable is tossed between the one in job archive and the one at stage ending. Next commit will fix this by just searching for it, as it'll be a sibling to the back button.

File: Modules/OpenGhostDir.cs (hunk 6 above)
Comment (stxticOVFL) [reply Modules/OpenGhostDir.cs:93]: See above.

File: Modules/OpenGhostDir.cs (hunk 5 above)
Comment (stxticOVFL) [reply Modules/OpenGhostDir.cs:77]: Same as above.

File: Modules/OpenGhostDir.cs (hunk 7 above)
Comment (stxticOVFL) [reply Modules/OpenGhostDir.cs:69]: Can't find anything from a quick search, `Exception` or any extensions to it don't contain this method.

File: Modules/OpenGhostDir.cs (hunk 3 above)
Comment (stxticOVFL) [reply Modules/OpenGhostDir.cs:36]: Used to be suffixed by a sensitive `else`, fixed in local!

File: Modules/OpenGhostDir.cs (hunk 2 above)
Comment (stxticOVFL) [reply Modules/OpenGhostDir.cs:31]: Will unwrap this later, this was just to catch issues I was having and I forgot to remove LOL

File: Modules/OpenGhostDir.cs (hunk 1 above)
Comment (stxticOVFL) [reply Modules/OpenGhostDir.cs:20]: Resolved in local.

**Comment by stxticOVFL** [comment]
I've moved the button to be above the back button, which is far more natural: 
image
All of the merge issues have been addressed.
There is also now a system that allows the back button to have some new added buttons to it; this can be separated if wanted, but is widely in-place for much more potential regarding cross-mod compatibility without having to concern anything about even talking to other mods directly. The back button provides an excellent interface for situational buttons (like this one) that have the very consistent back button present in their menus.

I would like to also add that the reason the code was so complex is not just because of how I program (because, admittedly, I tend to micro-optimize everything and that gives the clear appearance of it being so complex), but because it was very necessary in order to create a button that is inline with the rest of the game. 
In a common, widely used mod like NeonLite, there is nothing but detriment to not going the extra mile to give the UI a clean appearance that is consistent with the rest of the game, unlike NeonEraser where just adding a small button was all that was necessary. 
That is why I decided to go the path you've seen in the PR: the complexity wasn't for nothing, and was instead a result of said in-depth analysis. The new commit has just allowed me to build a better system using what I had learned previously, combined with the fact that I just simply didn't think of using the back button to my advantage as I didn't think of putting it there in the first place.

**Comment by MOPSKATER** [comment]
In a community project with more that one dev it's necessary to write simple to read code that causes as less sideeffects as possible. The main reason why I won't merge your stuff is that you are patching 6 methods to just create one button. That's neither lite nor simple. Please overthink your approach to this task.

**Comment by MOPSKATER** [comment]
Sorry but I really have not much time so I quickly rewrote that feature by myself

File: Modules/OpenGhostDir.cs [hunk 8]
```diff
@@ -10,50 +11,152 @@ internal class OpenGhostDir : Module
...
+        [HarmonyPrefix]
+        [HarmonyPatch(typeof(Singleton<BackButtonAccessor>), "OnDestroy")]
+        public static bool OnBackDestroy()
+        {
+            // the back button's singleton is EXTREMELY fragile
+            // so we just tell it to not worry
+            // (this also lets us destroy the copied backbutton without worry, making it safer)
+            // will this affect other singletons? idk
+            // it *shouldn't* bc this is supposed to never happen in the first place
+            return false;
```
Comment (MOPSKATER) [inline Modules/OpenGhostDir.cs:148]: Just don't do that. You don't know what could happen

**Review by MOPSKATER — COMMENTED** [review]
Like I already mentioned, there is just too much going on for such a tiny feature. This code could cause unexpected sideeffects

**Pull request closed without merging**